lib/scope.js

```diff
@@ -1792,6 +1808,27 @@ module.exports = {
       await scope.setFields( scope, { var_data, ensure_navigation: false, ensure_all_vars_used: true });
     },  // Ends set_regular_var()
 
+    set_secret_var: async ( scope, var_name, set_to_env_name ) => {
+      /** Sets a non-story table variable to a "secret" value (i.e. an env var) */
+      let set_to = process.env[set_to_env_name];
+      if ( set_to === undefined ) {
+        let set_to_env_msg = `The GitHub SECRET "${ set_to_env_name }" doesn't exist. Check for a typo and check your workflow file.`
+        await scope.addToReport( scope, { type: `error`, value: set_to_env_msg })
```

Comment:
If the secret doesn't exist, should we throw an error here? It's possible that the field will require text input, in which case the test might be able to move on just entering 'undefined' or some such. Do we want to be that permissive?

Reply:
Just a note that I think the bugs would end up being harder for the developer to track down. This will tell them right where to look for this particular mistake as opposed to triggering some other error further down the line.

Reply:
Haha, I know. I was kinda joking, I agree throwing is the much clearer option.